### ISSUE TITLE
Clarify / document capacity vs. number of segments

### ISSUE THREAD

**Issue by snazy** [snazy/ohc#40]
Document memory and segment layout for the various implementations and eviction algorithms.

**Comment by ben-manes** [comment]
You might also consider decoupling the eviction policy from the hash table segments. That could improve utilization, but its a trade-off of shifting complexities. See [HighScalability article](http://highscalability.com/blog/2016/1/25/design-of-a-modern-cache.html)) regarding concurrency scheme that allows that.

**Comment by snazy** [comment]
During the initial OHC design we decided to handle each segment individually - i.e. nothing is shared between segments - to keep exclusive accesses to a segment very short.

**Comment by ben-manes** [comment]
True. It does require exclusive access for a read which causes contention. I agree it may not be a performance problem due to other factors, e.g. deserialization overhead. The scheme outline above reduces contention by separating concerns using queuing theory.

**Comment by oobles** [comment]
I'd be very interested in your segment layout documentation when ready.  One thing I'm interested in doing is modifying the segments to allow dropping of individual segments (if the layout allows it). This is to be used in a clustered arrangement on multiple nodes.  A sort of Hazelcast style arrangement.  Dropping and adding a segments based on nodes being dropped/added.
